### ISSUE TITLE
Changing schema from scalar to list type can't find existing value

### ISSUE THREAD

**Issue by neilpa** [dgraph-io/dgraph#2060]
I was experimenting with modifying schemas and found an issue with changing from a scalar type to a list (at least for strings). After modifying the schema AND adding additional values, `eq` queries for the initial value no long return results. For example

Create the initial predicate
```
pred: string @index(exact) .
```

Add some data as normal
```
{
  set {
    <0x1234> <pred> "first" .
  }
}
```

Then change to a list type
```
pred: [string] @index(exact) .
```

Expected results are returned still w/ this query
```
{
  q(func: eq(pred, "first")) {
    uid
    pred
  }
}
```

But adding an additional value
```
{
  set {
    <0x1234> <pred> "second" .
  }
}
```

And then querying again

```
{
  q1(func: eq(pred, "first")) {
    pred
  }
  q2(func: eq(pred, "second")) {
    pred
  }
}
```

... fails when searching for the original value but not subsequent ones

```
{
  "data": {
    "q1": [],
    "q2": [
      {
        "pred": [
          "second",
          "first"
        ]
      }
    ]
  }
}
```

- What version of Dgraph are you using?

```
Dgraph version   : v1.0.1
Commit SHA-1     : ebce3481
Commit timestamp : 2017-12-20 13:59:58 +0530
Branch           : HEAD
```

- What is the hardware spec (RAM, OS)?

DGraph docker image

**Comment by pawanrawal** [comment]
Sounds like a bug to me.

**Comment by neilpa** [comment]
One more thing I noticed, `delete` mutations aren't working as expected either

```
{
  delete {
    <0x1234> <pred> "first" .
  }
}
```

The above gets a successful response

```
  q1(func: eq(pred, "second")) {
    pred
  }
```

But it's still showing up in the results

```
{
  "data": {
    "q1": [],
    "q2": [
      {
        "pred": [
          "second",
          "first"
        ]
      }
    ]
  }
}
```

**Comment by pawanrawal** [comment]
I have submitted a fix in 629f3a7ba0b18447ccfcecb7dd9c0a0ca26d1d96 and it would be part of the next release.
